### ISSUE TITLE
Two parameters

### ISSUE THREAD

**Issue by ErikOsinga** [tomcharnock/information_maximiser#7]
Hi,

I have tried to use this code to estimate the mean and variance of a one-dimensional Gaussian. I have estimated these separately successfully, but when I try to train the network with the two parameters at the same time it does not seem to work. 

I think the problem is that the determinant of the Fisher matrix is orders of magnitude different for the training and the test set, as shown in the figure below.

![variables_vs_epochs](https://user-images.githubusercontent.com/12582441/48078291-6866d780-e1e9-11e8-8fc9-c7acb9a45892.png)

Perhaps I am giving wrongly shaped data as input for the calculation of the derivatives for the network, as I could not find anywhere how to give the 'upper' and 'lower' perturbed data to the network.

Since the derivative denominator has to be given in a numpy array of shape [number of parameters]. I assumed the upper and lower perturbed training images for two parameters theta1 and theta2 should be given in a numpy array of shape 

t_p = (number_of_simulations, number_of_parameters, number_of_datapoints)

Thus it will be for every simulation a 2D array containing the perturbed theta1 simulations and the perturbed theta2 simulations [ t_p(theta1),t_p(theta2) ].
I have done this equivalently for t_m.

If this is the correct way to feed the network the perturbed simulations then I am lost as to why this is not working for 2 parameters at the same time.
Thanks in advance.

**Comment by ErikOsinga** [comment]
If this helps, I have (a portion of) the code I use to generate this plot available at 

https://gist.github.com/ErikOsinga/eacaaca8fb5b0f6cb5a4a8f98ba96856

**Comment by tomcharnock** [comment]
Hi Erik, 

It turned out that my example was using the v2_correlation_inversion branch. I've now updated the master branch so that it should work. I think the problem you were having was just an unfortunate choice of parameters for the Stochastic Gradient Descent Optimiser. v2 now uses Adam (again) which was the only difference between our two tests.

I've also uploaded my copy of the notebook so you can run that exactly (in v2). I do the MLE and the Fisher forecasts and they all look good so I think it should be fine now.

**Comment by ErikOsinga** [comment]
Many thanks for your response Tom. It seems that with ADAM the training proceeds a lot more stable, and it is working now.

**Comment by tomcharnock** [comment]
Fantastic, good to know!
